### PR TITLE
`JSONSerializable` using hex for `[]byte` (Plan B)

### PR DESCRIPTION
**Pull request by pinebit**

### CODE REVIEW

File: core/cbor/cbor.go [hunk 1]
```diff
@@ -2,20 +2,19 @@ package cbor
...
 )
 
-// ParseDietCBOR attempts to coerce the input byte array into valid CBOR
-// and then coerces it into a JSON object.
+// ParseDietCBOR attempts to coerce the input byte array into valid CBOR.
 // Assumes the input is "diet" CBOR which is like CBOR, except:
 // 1. It is guaranteed to always be a map
 // 2. It may or may not include the opening and closing markers "{}"
-func ParseDietCBOR(b []byte) (models.JSON, error) {
+func ParseDietCBOR(b []byte) (interface{}, error) {
```
Comment (samsondav) [inline core/cbor/cbor.go:17]: Would the return type not be `map[string]interface{}`?

File: core/services/pipeline/common.go [hunk 2]
```diff
@@ -237,25 +237,11 @@ func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 
 // MarshalJSON implements custom marshaling logic
 func (js JSONSerializable) MarshalJSON() ([]byte, error) {
-	if !js.Valid {
+	if !js.Valid || js.Val == nil {
```
Comment (samsondav) [inline core/services/pipeline/common.go:240]: Will this lead to literal `null` string being written to the database?

**Review by samsondav — DISMISSED** [review]

File: core/services/pipeline/task_params.go [hunk 3]
```diff
@@ -232,25 +230,32 @@ type BytesParam []byte
...
+	case ObjectParam:
+		if v.Type == StringType {
+			*b = BytesParam(v.StringValue)
+			return nil
+		}
+	case *ObjectParam:
+		if v.Type == StringType {
+			*b = BytesParam(v.StringValue)
+			return nil
+		}
```
Comment (jmank88) [inline core/services/pipeline/task_params.go:257]: Should an error be returned if not `StringType`?

File: core/cbor/cbor.go (hunk 1 above)
Comment (pinebit) [reply core/cbor/cbor.go]: Fixed

File: core/services/pipeline/common.go (hunk 2 above)
Comment (pinebit) [reply core/services/pipeline/common.go]: Correct, this is valid JSON and this is handled by `UnmarshalJSON`, producing `{Val: nil, Valid: false}`.

File: core/services/pipeline/task_params.go (hunk 3 above)
Comment (pinebit) [reply core/services/pipeline/task_params.go:256]: Oh such a great finding! Fixed and also checked all other *Params.

**Comment by jmank88** [comment]
Flake w/ confused message?? Or did some types change?
> === CONT  TestFluxMonitor_Deviation/legacy_mode
    assertion_compare.go:313: 
        	Error Trace:	integrations_test.go:610
        	Error:      	"100" is not greater than "60"
        	Test:       	TestFluxMonitor_Deviation/legacy_mode
        	Messages:   	[]

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**